File: modules/dms-vision/ios/GazeNetStub/GazeNet.swift
// The gaze network on a build WITHOUT DMS_GAZE_NET=1 (every production build): it does not exist.
// The podspec compiles this file only when the switch is off, and GazeNet/GazeNet.swift otherwise.
// Foundation only.

import Foundation

protocol GazeNetRunner: AnyObject {
  /// One batch-1 pass: cloud (1434), context (7), validity (478) → gaze (3), rotation (9).
  func run(cloud: [Float], context: [Float], validity: [Float]) throws -> (gaze: [Double], rotation: [Double])
  func close()
}

enum GazeNetFactory {
  static let available = false
  static let onnxRuntimeVersion: String? = nil

  /// The model's sha256 (diagnostics); nil without the net.
  static func modelSha256() -> String? { return nil }

  static func make() throws -> GazeNetRunner? { return nil }
}

File: modules/dms-vision/android/src/nogazenet/java/expo/modules/dmsvision/GazeNet.kt
// The gaze network on a build WITHOUT DMS_GAZE_NET=1 (every production build): it does not exist.
// build.gradle compiles this source set only when the switch is off, and src/gazenet/java otherwise.
// No model, no ONNX Runtime.

package expo.modules.dmsvision

import android.content.Context

object GazeNetFactory {
  const val available = false
  val onnxRuntimeVersion: String? = null

  /** The model's sha256 (diagnostics); null without the net. */
  @Suppress("UNUSED_PARAMETER")
  fun modelSha256(context: Context): String? = null

  @Suppress("UNUSED_PARAMETER")
  fun make(context: Context): GazeNetRunner? = null
}
